0001-interledger-architecture/0001-interledger-architecture.md

```diff
@@ -1,136 +1,168 @@
 `g.us.acmebank.acmecorp.sales.199` or `g.crypto.bitcoin.1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2`.
 
-When initiating an Interledger payment, the sender sends an ILP `prepare` packet to the connector. The packet is a binary message that includes the destination account, amount, condition, timeout, and transport-layer data for the receiver. The packet is relayed by connectors.
+For more information on the format of ILP addresses and how they apply to routing, see [IL-RFC-15: ILP Addresses](../0015-ilp-addresses/0015-ilp-addresses.md).
 
-### Ledger Layer
+If two parties in the Interledger have a "parent/child" connection rather than a peer-to-peer connection, the child can request an Interledger address that is under the parent's address in the hierarchy. For more information, see [IL-RFC-31: Interledger Dynamic Configuration Protocol (ILDCP)](0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md).
 
-In order to move packets from one party (sender, receiver or connector) to another, the parties need a facility for data transmission and, optionally, money transmission for settlement. These facilities may be provided by different protocols belonging to the so-called ledger layer. Which protocol is used between two parties is up to those parties to choose.
+## Interledger Security
+
+**Interledger uses *conditional transfers* to secure payments across multiple hops and even through untrusted connectors.** Everyone only needs to trust their direct peers, no matter how many connectors are involved in forwarding a given packet. Connectors take some risk, but this risk can be managed and is primarily based upon the connector's chosen peers.
+
+> **Hint:** Conditional transfers or *authorization holds* are the financial equivalent of a [two-phase commit](http://foldoc.org/two-phase%20commit).
+
+Because each party is isolated from risks beyond their immediate peers, longer paths are not inherently more risky than shorter paths. This unlocks the potential for many new ways of conveying money from any given sender to any given receiver.
+
+The original Interledger whitepaper presented an _atomic mode_ with a stronger guarantee of atomicity, but implementing atomic mode requires extra agreements that cannot be generalized to an open network. Specific subsets of an Interledger payment can upgrade to atomic mode if they agree on "notaries" to arbitrate disputes, but the specifications in the Interledger protocol suite are concerned primarily with what the whitepaper calls _universal mode_.
+
+### Conditions and Fulfillments
+
+Interledger uses the digest of the [SHA-256](https://en.wikipedia.org/wiki/SHA-2) hash function as the condition for prepare packets. (This choice was inspired by the design of the [Lightning Network](http://lightning.network).) The fulfill packet contains a valid 32-byte preimage for the hash specified in the prepare packet. Connectors are responsible for validating fulfillments in the Interledger Protocol layer.
```

Comment:
Conditions and Fulfillments were not inspired by Lightning. They started out as crypto-conditions and were simplified to just be SHA256 hashes to avoid network segmentation.

It's a happy coincidence that Lightning uses the same approach

Reply:
The decision to switch actually was partially inspired by a) a desire to be compatible with Lightning b) a conversation with @josephpoon who made the point that hashlocks can handle at least 80% of the use cases that crypto conditions could, but with far less than 20% of the effort and c) the realization that crypto conditions could lead to network fragmentation if different people supported different subsets of options